op-node/rollup/sequencing/origin_selector.go

```diff
@@ -238,3 +206,63 @@ func (los *L1OriginSelector) reset() {
 	los.currentOrigin = eth.L1BlockRef{}
 	los.nextOrigin = eth.L1BlockRef{}
 }
+
+var (
+	ErrInvalidL1Origin      = fmt.Errorf("origin-selector: currentL1Origin.Hash != l2Head.L1Origin.Hash")
+	ErrNextL1OriginOrphaned = fmt.Errorf("origin-selector: nextL1Origin.ParentHash != currentL1Origin.Hash")
+	ErrNextL1OriginRequired = fmt.Errorf("origin-selector: nextL1Origin not supplied but required to satisfy constraints")
+)
+
+// FindL1OriginOfNextL2Block finds the L1 origin of the next L2 block.
+// It returns an error if there is no way to build a block satisfying
+// derivation constraints with the supplied data.
+// You can pass an empty nextL1Origin if it is not yet available
+// removing the need for block building to wait on the result of network calls
+// This method is designed to be pure (it only reads the cfg property of the receiver)
+// and should not have any side effects.
+func (los *L1OriginSelector) findL1OriginOfNextL2Block(
+	l2Head eth.L2BlockRef,
+	currentL1Origin eth.L1BlockRef, nextL1Origin eth.L1BlockRef,
+	matchAutoDerivation bool) (eth.L1BlockRef, error) {
+
+	if (currentL1Origin == eth.L1BlockRef{}) {
+		panic("origin-selector: currentL1Origin is empty")
```

Comment:
Could this ever happen when launching from genesis, launching from a dev-L1, or something like that?

Reply:
Good question, I will l dig into this. Anyway, it seems prudent here to just return `ErrInvalidOrigin` instead of panicking.